### ISSUE TITLE
twisted/test/test_stateful.py fails on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8718]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8718|
|Type|enhancement|
|Created|2016-08-02 11:31:16Z|
|Branch|https://github.com/twisted/twisted/tree/8718-rodrigc-stateful-py3|

If I try to run:

tox -e py35-tests twisted/test/test_stateful.py

it fails

<details><summary>Searchable metadata</summary>

```
trac-id__8718 8718
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8718_rodrigc_stateful_py3 8718-rodrigc-stateful-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1470137476434758 1470137476434758
changetime__1470656475226602 1470656475226602
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/472

**Comment by twisted-trac** [comment]
|[<img alt="acabhishek942's avatar" src="https://avatars.githubusercontent.com/u/7778186?s=50" width="50" height="50">](https://github.com/acabhishek942)<a name="note_2"></a>|@acabhishek942 commented|
|-|-|

changes looks good.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset d3bb36fa54cecd84404f53ea375df9843cf6d90c
```
#!CommitTicketReference repository="" revision="d3bb36fa54cecd84404f53ea375df9843cf6d90c"
Merge pull request #472 from twisted/8718-rodrigc-stateful-py3

Author: rodrigc
Reviewer: acabhishek942
Fixes: #8718

Fix twisted.test.test_stateful on Python 3
```
